lib/spack/spack/platforms/_platform.py

```diff
@@ -143,3 +145,6 @@ def oses():
             for o in sorted(self.operating_sys.values()):
                 yield o._cmp_iter
         yield oses
+
+    def is_64bit(self):
+        return platform.machine().endswith('64')
```

Comment:
This is problematic because it's executing code to determine the bit support for the current operating context of Python whereas a `Platform` can describe a context other than the current one. This must either

* be determined from the existing attributes of the `Platform` object
* be passed in to `__init__`

Also I think some systems would report `ppc64le` so this check wouldn't cover all cases. Since this is only needed for Windows my guess is

* you can raise a `NotImplemented` exception here
* only actually implement this in the `Windows` platform

Reply:
Right, `Platform` is used in spec, this wouldn't return the correct result. Will refactor. This change was originally requested when this PR was first made due to a review comment suggesting there is more general use for this type of function, with that in mind, and your comment about limiting this to Windows, do you see a more general use here?